### PR TITLE
Reactive Tesla Armour

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Blocks hits, spits lightning back (just damage, no stun), admin spawn only. You know the drill by now.

:cl: Kor
rscadd: Adds reactive tesla armour. Adminspawn only.
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/armor.dm [hunk 1]
```diff
@@ -221,6 +216,22 @@
...
+	if(!active)
+		return 0
+	if(prob(hit_reaction_chance))
+		owner.visible_message("<span class='danger'>The [src] blocks the [attack_text], sending out arcs of lightning!</span>")
+		for(var/mob/living/M in view(6, owner))
+			if(M == owner)
+				continue
+			owner.Beam(M,icon_state="purple_lightning",icon='icons/effects/effects.dmi',time=5)
+			M.adjustFireLoss(25)
+			playsound(M, 'sound/machines/defib_zap.ogg', 50, 1, -1)
```
Comment (Dorsisdwarf) [inline code/modules/clothing/suits/armor.dm]: Is there a way to make it do as much damage as the hit would have done?

**Comment by WJohn** [comment]
Neat :+1: 
More inaccessible content is cool, it gives special things that people will want for away missions or space content.

File: code/modules/clothing/suits/armor.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/clothing/suits/armor.dm]: There is but that'd mean tasers would be weird

**Comment by GunHog** [comment]
I imagine this will eventually be added into play somehow, in addition to the others. While these are obviously high tech and something that Science would be able to make (The RD gets one at roundstart, after all), there is a group of people who would oppose any additional content to Science in any shape form or fashion (barring nerfs), so Away Mission/Cargo or Antags are probably how they will be introduced.

**Comment by KazeEspada** [comment]
Nice, I tell you the idea and bam you make it.

**Comment by Jazaen** [comment]
Well, lots of content that was already added is gated behind science, and currently they depend on no department other than mining (and, rarely, not even that).
Maybe add a mech version, so we can sneak in more Fallout 4 memes.

**Comment by Anonus** [comment]
Don't use purple lightning, use the red lightning or goof's tesla lightning when that gets in.

**Comment by Dorsisdwarf** [comment]
Hey just a pointer don't add all the adminspawn stuff to R&D.

**Comment by KorPhaeron** [comment]
@Dorsisdwarf 

The fucking phazon ended up in R&D nothing is sacred

**Comment by Jazaen** [comment]
At least phazon is gated behind research and anomaly core, so you need to leave your warm nest and hunt some anomalies.
Hm, what about adding more uses to anomaly cores other than research/mechs?

**Comment by Dorsisdwarf** [comment]
Make anomaly cores activated by attatched signaller. Instant bluespess

**Comment by Jordie0608** [comment]
Is this ready?

**Comment by KorPhaeron** [comment]
@Jordie0608

Yes, but I dont mind waiting till tesla gets merged for prettier lightning

**Comment by Dorsisdwarf** [comment]
Tesla is merge conflicting

**Comment by KorPhaeron** [comment]
Just merge it then I'll change the lightning colours later

**Comment by optimumtact** [comment]
update to latest master
git rebase -Xignore-all-whitespace upstream/master

**Comment by Iamgoofball** [comment]
imo redo this post-tesla engine so you can use tesla_zap()

**Pull request closed without merging**